ludwig/schema/model_config.py

```diff
@@ -0,0 +1,529 @@
+            if self.combiner.type != upgraded_config_dict[COMBINER][TYPE]:
+                self.combiner = combiner_registry.get(upgraded_config_dict[COMBINER][TYPE]).get_schema_cls()()
+
+            if self.combiner.type == SEQUENCE:
+                encoder_family = SEQUENCE
+            else:
+                encoder_family = None
+            self._set_attributes(self.combiner, upgraded_config_dict[COMBINER], feature_type=encoder_family)
+
+        # ===== Trainer =====
+        if TRAINER in upgraded_config_dict:
+            self._set_attributes(self.trainer, upgraded_config_dict[TRAINER])
+
+        # ===== Global Preprocessing =====
+        if PREPROCESSING in upgraded_config_dict:
+            self._set_attributes(self.preprocessing, upgraded_config_dict[PREPROCESSING])
+
+        # ===== Hyperopt =====
+        self.hyperopt = upgraded_config_dict.get(HYPEROPT, {})
+        self._set_hyperopt_defaults()
+
+        # ===== Validate Config =====
+        self._validate_config(self.to_dict())
+
+    def __repr__(self):
+        config_repr = self.to_dict()
+        config_repr[INPUT_FEATURES] = self.input_features.filter_features()
+        config_repr[OUTPUT_FEATURES] = self.output_features.filter_features()
+        config_repr[DEFAULTS] = self.defaults
+        return yaml.dump(config_repr, sort_keys=False)
+
+    @classmethod
+    def from_dict(cls, dict_config):
+        return cls(dict_config)
+
+    @classmethod
+    def from_yaml(cls, yaml_path):
+        with open(yaml_path) as stream:
+            try:
+                yaml_config = yaml.safe_load(stream)
+            except yaml.YAMLError as e:
+                raise yaml.YAMLError(f"Cannot parse input yaml file: {e}")
+        return cls(yaml_config)
+
+    @staticmethod
+    def _set_feature_column(config: dict) -> None:
+        for feature in config[INPUT_FEATURES] + config[OUTPUT_FEATURES]:
+            if COLUMN not in feature:
+                feature[COLUMN] = feature[NAME]
+
+    @staticmethod
+    def _set_proc_column(config: dict) -> None:
+        for feature in config[INPUT_FEATURES] + config[OUTPUT_FEATURES]:
+            if PROC_COLUMN not in feature:
+                feature[PROC_COLUMN] = compute_feature_hash(feature)
+
+    @staticmethod
+    def _upgrade_config(config_dict: dict) -> dict:
+        """Helper function used to run backwards compatibility check on the config and return an upgraded version.
+
+        Args:
+            config_dict: Config Dictionary
+        """
+        return upgrade_config_dict_to_latest_version(config_dict)
+
+    @staticmethod
+    def _validate_config(config_dict: dict) -> None:
+        """Helper function used to validate the config using the Ludwig Schema.
+
+        Args:
+            config_dict: Config Dictionary
+        """
+        validate_config(config_dict)
+
+    @staticmethod
+    def _get_config_cls(section: str, section_type: str, feature_type: str) -> BaseMarshmallowConfig:
```

Comment:
Might rename this to `_get_config_nested_cls` to be clear at a glance

Reply:
Good call!